### DISCUSSION TITLE
Guidelines and Process for Contribution

### DISCUSSION THREAD

**Discussion by mchoccie** [danielmiessler/Fabric#658] [Q&A]
Hi all, I'm new to Fabric and the world of open source. I'm extremely interested in the concept of Fabric and would certainly like to contribute a pattern of interest. Here is my question: Are there any rules/guidelines/steps I must follow before I can begin contributing. e.g. do I need to fork the repo first or can I clone the repo, make a change then make a pull request? Any kind of guidance into how this works would be much appreciated. I have had a look at the discussion thread and haven't found anything that matches my query. But if there is, and I've overlooked it, I'd be happy to be pointed in the right direction.

Thanks all.

**Comment by AO12612** [comment]
seconded.

Maybe some guideline documentation would help ( I read about the reimplementation in go, so maybe that needs to happen first ?)

**Comment by Wayku7** [comment]
I was wondering the same things
